test/tools/fuzzTesting/fuzzHelper.h

```diff
@@ -47,14 +48,20 @@ class RandomCode
 	static RandomCodeBase& get()
 	{
 		static RandomCode instance;
-		return instance.generator;
+		return *instance.generator;
 	}
 	RandomCode(RandomCode const&) = delete;
 	void operator=(RandomCode const&) = delete;
+	~RandomCode() { delete generator; }
 
 private:
-	RandomCode(){}
-	BoostRandomCode generator;
+	RandomCode(){
+		if (Options::get().randomTestSeed.empty())
+			generator = new BoostRandomCode();
```

Comment:
This abstraction is stupid. Only use DeterministicRandomCode (with better name).
If seed is empty just pick a random one (you can use random_device for it).

Reply:
this is just an example. there will be a different implementation of DeterministicRandomCode